include/dfu/pcd.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/** @file pcd.h
+ *
+ * @defgroup pcd Peripheral Core DFU
+ * @{
+ * @brief API for handling DFU of peripheral cores.
+ *
+ * The PCD API provides functions for sending/receiving DFU images between
+ * cores on a multi core system. The cores communicate through a command
+ * structure (CMD) which is stored in a shared memory location.
+ *
+ */
+
+#ifndef PCD_H__
+#define PCD_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include <device.h>
+
+#ifdef CONFIG_SOC_SERIES_NRF53X
+
+/* These must be hard coded as this code is preprocessed for both net and app
+ * core.
+ */
+#define APP_CORE_SRAM_START 0x20000000
+#define APP_CORE_SRAM_SIZE KB(512)
+#define RAM_SECURE_ATTRIBUTION_REGION_SIZE 0x2000
+#define PCD_CMD_ADDRESS (APP_CORE_SRAM_START \
+			+ APP_CORE_SRAM_SIZE \
+			- RAM_SECURE_ATTRIBUTION_REGION_SIZE)
+#endif
+
+
+/** @brief Opaque type */
+struct pcd_cmd;
+
+/** @brief Get a PCD CMD from the specified address.
+ *
+ * @param addr The address to check for a valid PCD CMD.
+ *
+ * @retval A pointer to the PCD CMD if successful.
+ *           Otherwise, NULL is returned.
+ */
+struct pcd_cmd *pcd_cmd_get(void *addr);
```

Comment:
If `addr` is an address, and you pass addresses to this function (e.g. `PCD_CMD_ADDRESS`) then it would be better to let this function take a `uint32_t`.

Reply:
Changed to `off_t` to follow zephyr convention when operating on these types of addresses/offsets.